go-controller/pkg/util/kube.go

```diff
@@ -712,7 +712,11 @@ func NoHostSubnet(node *v1.Node) bool {
 		return false
 	}
 
-	nodeSelector, _ := metav1.LabelSelectorAsSelector(config.Kubernetes.NoHostSubnetNodes)
+	nodeSelector, err := metav1.LabelSelectorAsSelector(config.Kubernetes.NoHostSubnetNodes)
+	if err != nil {
+		klog.Errorf("NoHostSubnetNodes label selector is not valid: %v", err)
```

Comment:
nit: same comment as with `func IsHybridOverlayNode(node *kapi.Node) bool {` add nodeInfo
and LOL these two util's are doing the same thing! :) we could remove one of them...